HRT-28 create course info

EricPedrido opened the pull request:
### Purpose

<!---
Describe the problem or feature.
Consider providing an overview of why the work is taking place (with any relevant links); don’t assume familiarity with the history.
-->

Adds front end for creating chart/providing course information just before starting the game.

[HRT-28](https://softeng761team5.atlassian.net/browse/HRT-28)
<!---
Link to the related issues [KEY-000](link)
e.g [HRT-27](https://softeng761team5.atlassian.net/browse/HRT-27)
--->
<!-- []() -->
<!-- []() -->

### Approach

<!---
How does this change address the problem
--->
- Adds a new page for creating course information
- Form is responsive to window size changes
- Added antd icons to the package. Not sure if it is added in the PR, but if you need to, then it's: 
```javascript 
npm install --save @ant-design/icons
```

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

<!--
If this PR contains a breaking change, describe the impact and migration path for existing applications below.
-->

### Screenshots before and after this change (required for UI changes)
#### Does this PR involve a UI change?

- [X] Yes
- [ ] No

#### Screenshot(s) of UI before this PR
n/a
<!---
Required if there is UI change
--->

#### Screenshot(s) of UI after this PR
![screenshot](https://user-images.githubusercontent.com/43208889/93061366-8fd13a00-f6c7-11ea-914f-347809090e65.PNG)
)

<!---
Required if there is UI change
--->

gargi-bansal committed:
Created and styled the homepage

gargi-bansal committed:
Created and styled the homepage

gargi-bansal committed:
Made requested changes

gargi-bansal committed:
Made requested changes

gargi-bansal committed:
Merge branch 'master' into 26-create-homepage

gargi-bansal committed:
Merge branch 'master' into 26-create-homepage

gargi-bansal committed:
Fixed test file

EricPedrido committed:
Add Temp Header and Text

App.tsx has been temporarily modified to launch to CourseInfo page for testing purposes

EricPedrido committed:
Add Temp Header and Text

App.tsx has been temporarily modified to launch to CourseInfo page for testing purposes

EricPedrido committed:
Merge remote-tracking branch 'origin/master' into 28-create-course-info-page

Commit:
feat(front-end): add form container and input boxes

App.tsx is modified to show CourseInfo.tsx by default for testing purposes

HRT-28

Commit:
feat(front-end): add form container and input boxes

App.tsx is modified to show CourseInfo.tsx by default for testing purposes

HRT-28

EricPedrido committed:
Merge 'origin/master' into HRT-28-create-course-info

Commit:
feat(courseinfo.tsx,.css): add buttons and fix spacing

HRT-28

Commit:
feat(courseinfo.tsx,.css): add buttons and fix spacing

HRT-28

EricPedrido committed:
Merge remote-tracking branch 'origin/master' into HRT-28-create-course-info

Commit:
fix(courseinfo.tsx,.css): make header the same as other headers

HRT-28

Commit:
fix(courseinfo.tsx,.css): make header the same as other headers

HRT-28

Commit:
feat(courseinfo.tsx,.css): add textbox prompts and dropdowns

Replace number inputs with dropdowns and add example placeholder texts for textboxes

HRT-28

Commit:
feat(courseinfo.tsx,.css): add textbox prompts and dropdowns

Replace number inputs with dropdowns and add example placeholder texts for textboxes

HRT-28

lucycy98 reviewed (changes requested):
hiii looks good 2 things: 
- can you please rebase with master? its a little out of date. 
- are we putting shadows for all page titles? e.g 'create chart'
- when I select an item for course age and cohort size, the text field should be filled with the value I select. currently it is just 'select'. see below:
![image](https://user-images.githubusercontent.com/34847224/93838111-a0436f00-fcdc-11ea-8cff-3c54277260a6.png)

EricPedrido commented:
> * can you please rebase with master? its a little out of date.

Yup will do

> * are we putting shadows for all page titles? e.g 'create chart'

This is how it was in the hifi prototypes, but i can remove the shadows for consistency with other pages

> * when I select an item for course age and cohort size, the text field should be filled with the value I select. currently it is just 'select'.

I will need to look into this, I've done it how they've documented on antd's website but I'll try find a fix somewhere online

EricPedrido committed:
Merge remote-tracking branch 'origin/HRT-28-create-course-info' into HRT-28-create-course-info

Commit:
fix(courseinfo.tsx/.css): fix dropdowns not showing selections

Changed dropdowns into antd select components

HRT-28

preetpatel commented on src/pages/CourseInfo.tsx:
```diff
@@ -0,0 +1,108 @@
+const CourseInfo: React.FC = () => {
+  const onConfirmClick = () => {
+    // TODO: Write code here to redirect to cards screen
+  };
```
Think the cards endpoint is now available :)

preetpatel commented on src/pages/CourseInfo.tsx:
```diff
@@ -0,0 +1,108 @@
+  };
+
+  const onCancelClick = () => {
+    // TODO: Write code here to redirect back to homepage
```
This can also be implemented now

EricPedrido committed:
Merge remote-tracking branch 'origin/master' into HRT-28-create-course-info

Commit:
feat(front-end): route play to courseinfo page and update UI style

Added courseinfo to the app.tsx, routed the play button on homepage to go to courseinfo, put
elements inside the form container

HRT-28

lucycy98 reviewed (approved):
thanks for the changes - LGTM

mitchfaulcon reviewed (approved):
Looks good to me, I'll implement the page routing when I do the play reason ticket
